core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/AbstractHealthCheckedEndpointGroupBuilder.java

```diff
@@ -184,6 +188,37 @@ public AbstractHealthCheckedEndpointGroupBuilder maxEndpointCount(int maxEndpoin
         return this;
     }
 
+    /**
+     * Sets the
+     * <a href="https://en.wikipedia.org/wiki/Basic_access_authentication">HTTP basic access authentication</a>
+     * header using {@link HttpHeaderNames#AUTHORIZATION}.
+     */
+    public AbstractHealthCheckedEndpointGroupBuilder auth(BasicToken token) {
```

Comment:
This isn't your fault, but I'm puzzled by having multiple `auth(...)` methods. Wouldn't it be possible to reduce the number of methods by making one interface to `XXXToken` classes? 🤔 

ex)
```java
interface AuthorizationToken {
  String asHeaderValue();
} 
```

Reply:
👍 It would be nice we have some factories in the `AuthorizationToken`.
```java
interface AuthorizationToken {
  static BasicToken ofBasicToken(String username, String password);
  String asHeaderValue();
}
 
public final class BasicToken implements AuthorizationToken {

}
```

Reply:
By the way, it seems like not a good idea. Make `AuthorizationToken` interface because the contract of `AuthorizationToken` is weak and has just `asHeaderValue()`.
How about making an abstract class and prohibit users from directly extends the class?
```
public abstract class AuthorizationToken {
  static BasicToken ofBasicToken(String username, String password);
  // ... add other tokens

  // Disallow to extends this class out of auth module.
  AuthorizationToken() {}

  public abstract String asHeaderValue();
}
```

Reply:
I think it's a better idea 👍 👍

Reply:
@ikhoon @heowc this has been addressed, please take a look

Reply:
Unfortunately, we could delete public APIs that cause binary incompatibility even though the new API is super type of them.

Reply:
@ikhoon Added back